src/main/scala/viper/gobra/frontend/Desugar.scala

```diff
@@ -166,8 +166,8 @@ object Desugar {
         case NoGhost(x: PConstDecl) => constDeclD(x)
         case NoGhost(x: PMethodDecl) => Vector(registerMethod(x))
         case NoGhost(x: PFunctionDecl) => Vector(registerFunction(x))
-        case x: PMPredicateDecl => Vector(mpredicateD(x))
-        case x: PFPredicateDecl => Vector(fpredicateD(x))
+        case x: PMPredicateDecl => Vector(registerMPredicate(x))
```

Comment:
Why wasn't it necessary to register predicates before but now it is?

Reply:
If I am not mistaken, Felix's encoding performs lookups for predicate proxys in the LookupTable available in the `Context` trait. That table is filled here. Before we had defunctionalization, we didn't perform such lookups.